perun-core/src/main/java/cz/metacentrum/perun/core/entry/VosManagerEntry.java

```diff
@@ -594,6 +595,96 @@ public void removeSponsorRole(PerunSession sess, Vo vo, Group group) throws Grou
 		AuthzResolverBlImpl.unsetRole(sess, group, vo, Role.SPONSOR);
 	}
 
+	@Override
+	public BanOnVo setBan(PerunSession sess, BanOnVo ban) throws PrivilegeException, MemberNotExistsException {
+		Utils.checkPerunSession(sess);
+
+		// We have to fetch the member object from DB because of authorization. The given ban object might contain
+		// invalid combination of voId and memberId
+		Member member = perunBl.getMembersManagerBl().getMemberById(sess, ban.getMemberId());
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "setBan_BanOnVo_policy", Collections.singletonList(member))) {
+			throw new PrivilegeException("setBan");
+		}
+
+		return vosManagerBl.setBan(sess, ban);
+	}
+
+	@Override
+	public void removeBan(PerunSession sess, int banId) throws PrivilegeException, BanNotExistsException {
+		Utils.checkPerunSession(sess);
+
+		BanOnVo banOnVo = perunBl.getVosManagerBl().getBanById(sess, banId);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "vo-removeBan_int_policy", Collections.singletonList(banOnVo))) {
+			throw new PrivilegeException("removeBan");
+		}
+
+		vosManagerBl.removeBan(sess, banId);
+	}
+
+	@Override
+	public void removeBanForMember(PerunSession sess, Member member) throws PrivilegeException, BanNotExistsException {
+		Utils.checkPerunSession(sess);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "vo-removeBanForMember_member_policy", Collections.singletonList(member))) {
+			throw new PrivilegeException("removeBanForMember");
+		}
+
+		Optional<BanOnVo> banOnVo = perunBl.getVosManagerBl().getBanForMember(sess, member.getId());
+
+		if (!banOnVo.isPresent()) {
+			throw new BanNotExistsException("Given member is not banned.");
+		}
+
+		vosManagerBl.removeBanForMember(sess, member.getId());
+	}
+
+	@Override
+	public BanOnVo getBanById(PerunSession sess, int banId) throws BanNotExistsException, PrivilegeException {
+		Utils.checkPerunSession(sess);
+
+		BanOnVo ban = vosManagerBl.getBanById(sess, banId);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "vo-getBanById_int_policy", Collections.singletonList(ban))) {
+			throw new PrivilegeException("getBanById");
+		}
+
+		return ban;
+	}
+
+	@Override
+	public BanOnVo getBanForMember(PerunSession sess, Member member) throws PrivilegeException {
```

Comment:
Consider adding GROUPADMIN privileges for get methods.

Reply:
I have added GROUPADMIN to all policies for getting vobans.